.github/workflows/ci-python.yml

```diff
@@ -0,0 +1,149 @@
+name: Python checks
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - compass/**
+      - .github/workflows/lint-python.yml
+      - pyproject.toml
+      - pixi.lock
+  pull_request:
+    paths:
+      - compass/**
+      - .github/workflows/lint-python.yml
+      - pyproject.toml
+      - pixi.lock
+  workflow_dispatch:
+
+jobs:
+  lint:
+    name: Lint Python Code Base with Ruff
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v5
+      - uses: astral-sh/ruff-action@v3
+        with:
+          version: "latest"
+          args: "check"
+          src: "./compass"
+      - uses: astral-sh/ruff-action@v3
+        with:
+          version: "latest"
+          args: "format --check"
+          src: "./compass"
+
+  unit-tests:
+    needs: lint
+    name: Python Unit Tests (Pixi)
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      max-parallel: 8
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v5
+      with:
+        fetch-depth: 0
```

Comment:
Do we need `fetch-tags`? Sometime ago this flag was a placeholder and not doing anything. I don't know if they finally activated that.

Reply:
If it works without it, just leave as it is.

The question is this working with the correct version or using a fallback version? If using a fallback version, would that mislead the cache, i.e. use an old version because it can't distinguish?